--- a/runit.c
+++ b/runit.c
@@ -18,9 +18,9 @@
 
 /* #define DEBUG */
 
-#define INFO "- runit: "
-#define WARNING "- runit: warning: "
-#define FATAL "- runit: fatal: "
+#define INFO "runit: "
+#define WARNING "runit: warning: "
+#define FATAL "runit: fatal: "
 
 const char * const stage[3] ={
   "/opt/puppetizer/etc/runit/1",
@@ -92,9 +92,6 @@
   if (RB_DISABLE_CAD == 0) reboot_system(0);
 #endif
 
-  strerr_warn3(INFO, "$Id: 25da3b86f7bed4038b8a039d2f8e8c9bbcf0822b $",
-               ": booting.", 0);
-
   /* runit */
   for (st =0; st < 3; st++) {
     /* if (st == 2) logwtmp("~", "reboot", ""); */
@@ -117,8 +114,6 @@
           dup2(ttyfd, 0);
           if (ttyfd > 2) close(ttyfd);
         }
-        else
-          strerr_warn2(WARNING, "unable to open /dev/console: ", &strerr_sys);
       }
       else
         setsid();
